### ISSUE TITLE
[Select] Some select components do not have visible label

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3104]
## Current behavior 
There is no external or internal visible label for examples on [documentation website](https://sanomalearning.design/categories/components/select/usage/) and for Storybook variants:

1. [Options Styling](https://storybook.sanomalearning.design/?path=/story/form-select--options-styling)
2. [No Visible Label](https://storybook.sanomalearning.design/?path=/story/form-select--no-visible-label)
3.  [Display Inline Block](https://storybook.sanomalearning.design/?path=/story/form-select--display-inline-block)

Having visible label can be very helpful especially for users with cognitive disabilities. It's also WCAG 3.3.2 requirement. 

<img width="385" height="104" alt="Image" src="https://github.com/user-attachments/assets/84658d2f-2367-4b6e-90aa-9935062bb6f2" />

<img width="479" height="219" alt="Image" src="https://github.com/user-attachments/assets/1d81e844-78f9-4582-b704-1d3897bec144" />

## Suggested solution

I think in documentation website we should present the most accessible way to use Select which is using visible label.

For Options Styling and Display Inline Block visible labels can be used as well but in that cases label can be also provided with not selectable option, that is being displayed when user doesn't choose anything. This option should explain what user can select inside select. Using first selectable option as only visible label for select is not sufficient. Example of that is used in [Custom Styling](https://storybook.sanomalearning.design/?path=/story/form-select--custom-styling).

For No Visible label Variant additional information about how to met WCAG with this component should be added. Whenever select with no labels is used in application it is important to make sure some visual cues, preferably using text, indicating select purpose are added in close proximity of that select. By doing this WCAG 3.3.2 would be met and component would be more accessible.

### 👤 Your name

Kacper 

### 🧢 Your Product/Team

SLDS

**Comment by a11ymiko** [comment]
Lists of elements without visible labels in storybook

### Incorrectly labelled ❌🏷️

- [x] 1. [Switch Custom Icons](https://storybook.sanomalearning.design/?path=/story/form-switch--custom-icons) Already has issue #3016 
- [x] 2. [Combobox All](https://storybook.sanomalearning.design/?path=/story/form-combobox--all) At least all comboboxes with tags should have added `aria-label` to `input role='combobox'`. Other comboboxes that have placeholder doesn't require `aria-label` but it can be added to them as well as it's improving their accessibility.
- [x] 3. All Combobox Multiple stories
- [x] 4. All Combobox Single stories
- [ ] 5. [Form field Combobox](https://storybook.sanomalearning.design/?path=/story/form-form-field--combobox) Issue already created #3010
- [x] 6. [Select Custom Styling](https://storybook.sanomalearning.design/?path=/story/form-select--custom-styling) This is described in this issue
- [x] 7. [Select Options With Styling](https://storybook.sanomalearning.design/?path=/story/form-select--options-styling) This issue
- [x] 8. [Select Display Inline Block](https://storybook.sanomalearning.design/?path=/story/form-select--display-inline-block) This issue
- [x] 9. [Select All](https://storybook.sanomalearning.design/?path=/story/form-select--all)

### Correctly labelled ✅🏷️
1. [Checkbox No Visible Label](https://storybook.sanomalearning.design/?path=/story/form-checkbox-checkbox--no-visible-label)
2. [Checkbox group No Label](https://storybook.sanomalearning.design/?path=/story/form-checkbox-checkbox-group--no-label)
3. [Form field Composite](https://storybook.sanomalearning.design/?path=/story/form-form-field--composite)
4. All search field stories
5. [Select No Visible Label](https://storybook.sanomalearning.design/?path=/story/form-select--no-visible-label)

**Comment by Diaan** [comment]
I've set the `for` from the label and the `aria-describedby` to the correct settings as per #3010, but still nvda doesn't read everything when in browse mode:

<img width="1138" height="985" alt="Image" src="https://github.com/user-attachments/assets/38eac338-36f8-4caa-94c6-6db45cf693c6" />

But a native select with a native label doesn't read the label either when you are using browse mode:

<img width="357" height="435" alt="Image" src="https://github.com/user-attachments/assets/20332f1c-7b31-4534-928a-c0dbd65fa756" />
